File: platform/platform-core/src/main/java/ifml2/platform/core/fx/BundlesTable.java
package ifml2.platform.core.fx;

import javafx.application.Platform;
import org.osgi.framework.Bundle;

public class BundlesTable extends AbstractBundleEntryTable {

    public BundlesTable() {
        super();
    }

    @Override
    void bundleAdded(final Bundle b) {
        Platform.runLater(new Runnable() {
            @Override
            public void run() {
                getEntries().add(BundleEntry.createBundleEntry(b));
            }
        });
    }

    @Override
    void bundleRemoved(final Bundle bundle) {
        Platform.runLater(new Runnable() {
            @Override
            public void run() {
                BundleEntry be = getBundleEntry(bundle);
                getEntries().remove(be);
            }
        });
    }

    @Override
    void bundleModified(final Bundle bundle) {
        Platform.runLater(new Runnable() {
            @Override
            public void run() {
                BundleEntry be = getBundleEntry(bundle);
                if (be!=null) {
                    be.fireBundleModified();
                }
            }
        });
    }

}

File: platform/platform-core/src/main/kotlin/ifml2/platform/core/fx/BundlesTable.kt
package ifml2.platform.core.fx

import javafx.application.Platform
import org.osgi.framework.Bundle

class BundlesTable : AbstractBundleEntryTable() {

    internal override fun bundleAdded(bundle: Bundle) {
        Platform.runLater { entries.add(BundleEntry.createBundleEntry(bundle)) }
    }

    internal override fun bundleRemoved(bundle: Bundle) {
        Platform.runLater {
            val be = getBundleEntry(bundle)
            entries.remove(be)
        }
    }

    internal override fun bundleModified(bundle: Bundle) {
        Platform.runLater {
            val be = getBundleEntry(bundle)
            be?.fireBundleModified()
        }
    }

}
